app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsWithTagDialogFragment.java

```diff
@@ -166,30 +175,34 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
             sc.setChecked(prefs.getBoolean(type, true));
             sc.setOnCheckedChangeListener((buttonView, isChecked) -> prefs.edit().putBoolean(type, isChecked).apply());
 
-            String messageStr = getString(R.string.ingredients_in_this_product_are, name.toLowerCase());
+            Spanned messageToBeShown = Html.fromHtml(getString(R.string.ingredients_in_this_product_are, name.toLowerCase()));
             AppCompatButton helpNeeded = getView().findViewById(R.id.helpNeeded);
             boolean showHelpTakePhoto = getArguments().getBoolean("photos_to_be_validated", false);
+            boolean hasAmbiguousIngredient = tag != null && ambiguousIngredient != null;
             boolean showHelpTranslate = tag != null && tag.contains("unknown");
             boolean showHelpExtract = showHelpTranslate && getArguments().getBoolean("missing_ingredients", false);
             AppCompatImageView image = getView().findViewById(R.id.image);
             if (showHelpTakePhoto) {
-                messageStr = getString(R.string.unknown_status_missing_ingredients);
+                messageToBeShown = Html.fromHtml(getString(R.string.unknown_status_missing_ingredients));
                 image.setImageResource(R.drawable.ic_add_a_photo_dark_48dp);
                 image.setOnClickListener(v -> goToAddPhoto());
                 helpNeeded.setText(Html.fromHtml(getString(R.string.add_photo_to_extract_ingredients)));
                 helpNeeded.setOnClickListener(v -> goToAddPhoto());
+            } else if (hasAmbiguousIngredient) {
+                messageToBeShown = Html.fromHtml(getString(R.string.unknown_status_ambiguous_ingredients, ambiguousIngredient));
```

Comment:
If `ambiguousIngredient` is null error is raised.

Reply:
`hasAmbiguousIngredient` is computed based on the following statement : 
 `boolean hasAmbiguousIngredient = tag != null && ambiguousIngredient != null;`
So if it null, the boolean variable would be false.

 Could you explain what needs to be done here?

Reply:
You're right, didnt' see the bool declaration. I would prefer if `hasAmbiguousIngredient`, `showHelpTranslate`, `showHelpTranslate`, `showHelpExtract` are inlined. What do you think?

Reply:
Retained showHelpTranslate as it is used in 2 places. Made rest of the conditions inline